src/components/ImageCarousel.vue

```diff
@@ -1,74 +1,71 @@
 import MockData from '../assets/MockData.JSON'
 import { Carousel, Slide, Navigation } from 'vue3-carousel'
 import 'vue3-carousel/dist/carousel.css'
 
-// Will move state to parent component or store
-const state = reactive({
-    currentSlide: 0,
-    selectedImages: []
-})
 
-// Gets invoked when thumbnail images are selected and does 2 things:
-// 1. toggles the status of thumbnail images to add and remove from state
-// 2. updates the current slide (displayed image) 
+const store = useStore()
+const currentSlide = ref(0)
+
+// Checking if image isSelected to either add or remove yellow borderline
+const isSelected = (item) => store.getters.isSelected(item)
+
+// Invoked any time an image is clicked
 const handleThumbnailClick = (item, index) => {
- // Toggles the selection of thumbnail iamages
- const existingIndex = state.selectedImages.findIndex(img => img.basefile_name === item.basefile_name)
-    if (existingIndex >= 0) {
-        // Remove the image from selectedImages if it's already there
-        state.selectedImages.splice(existingIndex, 1)
-    } else {
-        // Add the image to selectedImages if it's not already there
-        state.selectedImages.push(item)
-    }
-    // Update the current slide to the last image in the selectedImages array, or the clicked image if none are selected
-    if (state.selectedImages.length > 0) {
-        const lastSelectedImage = state.selectedImages[state.selectedImages.length - 1]
-        const lastSelectedIndex = MockData.findIndex(img => img.basefile_name === lastSelectedImage.basefile_name)
-        state.currentSlide = lastSelectedIndex
-    } else {
-        state.currentSlide = index
-    }
-}
+  // Dispatch an action to toggle the selection status of the clicked image
+  // This will add the image to the selectedImages array in the store if it's not there
+  // or remove it if it is already in the array
+  store.dispatch('toggleImageSelection', item)
 
-// Method to check if a thumbnail image is selected
-const isSelected = (item) => {
-    return state.selectedImages.some(selectedItem => selectedItem.basefile_name === item.basefile_name);
+  // Checking if there are any selected images after the toggle action
+  if (store.state.selectedImages.length > 0) {
+    // If there are selected images, then we find the last selected image
+    const lastSelectedImage = store.state.selectedImages[store.state.selectedImages.length - 1]
+    // Then we find the index of the last selected image in the MockData array
+    // This is used to set the current slide to the last selected image
+    const lastSelectedIndex = MockData.findIndex(img => img.basefile_name === lastSelectedImage.basefile_name)
+    // Set the current slide to the last selected image
+    currentSlide.value = lastSelectedIndex
+  } else {
+    // If no images are selected, set the current slide to the index of the clicked image
+    // This ensures that the carousel shows the clicked image
+    currentSlide.value = index;
```

Comment:
Is all of this commented code needed?

Reply:
Just to explain here so that Lloyd (and anyone who uses this) can understand while I wfh :) I can delete after it's been used and understood